core/terminal.go

```diff
@@ -102,79 +112,84 @@ func (container *Container) runTerminal(
 		return fmt.Errorf("failed to create service for interactive terminal: %w", err)
 	}
 
-	eg, egctx := errgroup.WithContext(ctx)
-
-	// forward a io.Reader to websocket
-	forwardFD := func(prefix []byte) func(r io.Reader) {
-		return func(r io.Reader) {
-			eg.Go(func() error {
-				for {
-					b := make([]byte, 512)
-					n, err := r.Read(b)
-					if err != nil {
-						if err == io.EOF {
-							return nil
-						}
-						return err
-					}
-					message := append([]byte{}, prefix...)
-					message = append(message, b[:n]...)
-					err = conn.WriteMessage(websocket.BinaryMessage, message)
-					if err != nil {
-						return err
-					}
-				}
-			})
-		}
+	bk := container.Query.Buildkit
+	term, err := bk.OpenTerminal(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to open terminal: %w", err)
 	}
 
+	fmt.Fprintf(term.Stderr, "Attaching interactive terminal: ")
+	idtui.DumpID(idtui.NewOutput(term.Stderr, termenv.WithTTY(true)), svcID)
```

Comment:
@vito does this look ok to you? is this the best way there is to represent "where we are"?

<img width="990" alt="Screenshot 2024-06-10 at 11 13 28 AM" src="https://github.com/dagger/dagger/assets/179503/61d138f1-27c5-4306-bcb8-bb251d97eaa0">